raiden/tests/mocked/test_matrix_transport.py

```diff
@@ -161,13 +197,70 @@ def test_sending_nonstring_body(  # pylint: disable=unused-argument
 def test_processing_invalid_message_json(  # pylint: disable=unused-argument
     mock_matrix, skip_userid_validation, message_input
 ):
-    room, event = make_message(overwrite_data=message_input)
+    room, event = make_message_text(overwrite_data=message_input)
     assert not mock_matrix._handle_sync_messages([(room, [event])])
 
 
 def test_processing_invalid_message_type_json(  # pylint: disable=unused-argument
     mock_matrix, skip_userid_validation
 ):
     invalid_message = '{"_type": "NonExistentMessage", "is": 3, "not_valid": 5}'
-    room, event = make_message(overwrite_data=invalid_message)
+    room, event = make_message_text(overwrite_data=invalid_message)
     assert not mock_matrix._handle_sync_messages([(room, [event])])
+
+
+@pytest.mark.parametrize("retry_interval", [0.01])
+def test_retry_queue_send_only_once(mock_matrix, record_sent_messages):
+    # Pretend the Transport greenlet is running
+    mock_matrix.greenlet = True
```

Comment:
~Can we write this test without a mock of the transport?~ humm, not a the moment, we have to fix that.

Reply:
Yeah, the level of interdependency between the Transport and RetryQueue is pretty bad. 
#5607 summarizes that nicely, thanks.